bare_header/sifive_hca_0_5_x.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef __BARE_HEADER_SIFIVE_HCA_0_5_x__H
+#define __BARE_HEADER_SIFIVE_HCA_0_5_x__H
+
+#include "bare_header/device.h"
+
+#include <cstdio>
+#include <regex>
+#include <set>
+
+class sifive_hca_0_5_x : public Device {
+public:
+  sifive_hca_0_5_x(std::ostream &os, const fdt &dtb)
+      : Device(os, dtb, "sifive,hca-0.5.(\\d+)") {}
+
+  void emit_defines() override {
+    emit_compat("sifive,hca");
```

Comment:
Since it seems like the HCA has a very unstable hardware API, shouldn't the compatible string reflect that? If versions after 0.5.x are incompatible, they must have a distinct compatible string from the 0.5.x HCA or the wrong driver for the HCA will be pulled in.

Reply:
According to Vincent A. that develops and maintains the HCA IP, the API will never change on minor fix related to the last digit of the version. 
if API change the version will increase the major or minor number, so it means we will produce another file to include to the bare_metal header generator.
To avoid to use a wrong API the define "METAL_SIFIVE_HCA_VERSION" is generated and check into the scl-metal repository.

Reply:
Oh whoops, I was looking elsewhere. I see above you're comparing with "sifive,hca-0.5.\d+". Carry on :)